packages/api/lambdas/verifyProviderSecretsMigration.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const KMS = require('@cumulus/aws-client/KMS');
+const { dynamodbDocClient } = require('@cumulus/aws-client/services');
+const { isNil } = require('@cumulus/common/util');
+
+const verifyProvider = async (provider) => {
+  if (provider.encrypted === true) {
+    try {
+      await KMS.decryptBase64String(provider.username);
+      await KMS.decryptBase64String(provider.password);
+    } catch (_) {
```

Comment:
The errors that are thrown don't give you a specific issue, they say "invalid key or maybe you don't have access to the key or ...". In this case, we're just using the Lambda function to stop a deployment that has _not_ yet run the migration script. My vote is to leave it the way it is.

Reply:
Hmm, but what if this script is only failing because of some weirdness with KMS or a deployment error that **we** caused? Wouldn't we want the error to specifically identify the problem?

Reply:
Updated in 404b9f4e7